heka/sandbox/decoders/extract_telemetry_dimensions.lua

```diff
@@ -163,10 +161,20 @@ local function process_json(msg, json, parsed)
         clientId = parsed.clientId
         update_field(msg.Fields, "clientId"  , clientId)
 
-        -- restructure the main ping message
-        if parsed.type == "main" then
```

Comment:
It's probably a good idea to do saved-session too, as long as it doesn't kill us in cpu costs